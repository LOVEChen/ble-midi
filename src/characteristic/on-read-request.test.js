/* eslint-env jest */
jest.mock('bleno/lib/characteristic');
jest.mock('../util/logger');

test('should onReadRequest(offset, callback)', () => {
  const mockLog = require('../util/logger').default;
  const onReadRequest = require('./on-read-request').default;
  const offset = jest.fn();
  const callback = jest.fn();
  onReadRequest(offset, callback);
  expect(mockLog.mock.calls).toMatchSnapshot();
  expect(mockLog.mock.calls.length).toEqual(1);
  expect(mockLog.mock.calls[0].length).toEqual(2);
  expect(callback.mock).toMatchSnapshot();
  expect(callback.mock.calls.length).toEqual(1);
  expect(callback.mock.calls[0].length).toEqual(2);
});
