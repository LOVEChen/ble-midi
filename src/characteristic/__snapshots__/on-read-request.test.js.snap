// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should onReadRequest(offset, callback) 1`] = `
Object {
  "calls": Array [
    Array [
      "onReadRequest",
      [Function],
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;

exports[`should onReadRequest(offset, callback) 2`] = `
Object {
  "calls": Array [
    Array [
      0,
      Buffer [],
    ],
  ],
  "instances": Array [
    undefined,
  ],
}
`;
